src/libmeasurement_kit/libevent/dns_impl.hpp

```diff
@@ -53,37 +54,66 @@ class QueryContext : public NonMovable, public NonCopyable {
     }
 };
 
+struct evdns_base_deleter {
+    void operator()(evdns_base* p) {
+        constexpr int fail_requests = 1;
+        evdns_base_free(p, fail_requests);
+    }
+};
+
+struct evaddrinfo_deleter {
+    void operator()(evutil_addrinfo* p) {
+        evutil_freeaddrinfo(p);
+    }
+};
+
+using evdns_base_uptr = std::unique_ptr<evdns_base, evdns_base_deleter>;
+using evaddrinfo_uptr = std::unique_ptr<evutil_addrinfo, evaddrinfo_deleter>;
```

Comment:
I appreciate the usage of RAII here. Note to self: As part of more significant refactoring of this file, I'd like the function modified by this diff to return a `Var<event_base>` (where `Var<>` is a `shared_ptr<>` wrapper will null pointer exception checks) with the proper deleter set (I've typically used a lambda rather than a struct for this).